components/edit_post/edit_post.tsx

```diff
@@ -0,0 +1,481 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useRef, useState} from 'react';
+import classNames from 'classnames';
+import {useIntl} from 'react-intl';
+
+import {Post} from 'mattermost-redux/types/posts';
+import {Emoji, SystemEmoji} from 'mattermost-redux/types/emojis';
+
+import {Constants, ModalIdentifiers} from 'utils/constants';
+import {
+    formatGithubCodePaste,
+    formatMarkdownTableMessage,
+    getTable,
+    isGitHubCodeBlock,
+} from 'utils/paste';
+import {postMessageOnKeyPress, splitMessageBasedOnCaretPosition} from 'utils/post_utils';
+import {isMac} from 'utils/utils';
+import * as Utils from 'utils/utils';
+
+import DeletePostModal from 'components/delete_post_modal';
+import EmojiPickerOverlay from 'components/emoji_picker/emoji_picker_overlay';
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import Textbox, {TextboxClass} from 'components/textbox';
+import EmojiIcon from 'components/widgets/icons/emoji_icon';
+import {isInternetExplorer, isSafari} from '../../utils/user_agent';
+
+type OpenModal = {
+    ModalId: string;
+    dialogType: typeof React.Component;
+    dialogProps: {
+        post: Post;
+        isRHS?: boolean;
+    };
+};
+
+export type Actions = {
+    addMessageIntoHistory: (message: string) => void;
+    editPost: (input: Partial<Post>) => Promise<Post>;
+    unsetEditingPost: () => void;
+    openModal: (input: OpenModal) => void;
+    setShowPreview: (newPreview: boolean) => void;
+    scrollPostListToBottom: () => void;
+}
+
+export type Props = {
+    canEditPost?: boolean;
+    canDeletePost?: boolean;
+    readOnlyChannel?: boolean;
+    channelId: string;
+    codeBlockOnCtrlEnter: boolean;
+    ctrlSend: boolean;
+    config: {
+        EnableEmojiPicker?: string;
+        EnableGifPicker?: string;
+    };
+    maxPostSize: number;
+    shouldShowPreview: boolean;
+    useChannelMentions: boolean;
+    editingPost: {
+        post?: Post;
+        postId?: string;
+        refocusId?: string;
+        title?: string;
+        isRHS?: boolean;
+    };
+    actions: Actions;
+};
+
+export type State = {
+    editText: string;
+    caretPosition: number;
+    postError: React.ReactNode;
+    errorClass: string | null;
+    showEmojiPicker: boolean;
+    renderScrollbar: boolean;
+    scrollbarWidth: number;
+    prevShowState: boolean;
+};
+
+const {KeyCodes} = Constants;
+
+const TOP_OFFSET = 0;
+const RIGHT_OFFSET = 10;
+
+const EditPost = ({editingPost, actions, ...rest}: Props): JSX.Element => {
```

Comment:
Really personal preference, so 0/5, but I would prefer the rest of the props to be called `props` instead of `rest`.

Reply:
I just followed the "convention" from the code-base here: we only use `...props` once but `...rest` several times ... I would have preferred `props` as well